File: algorithms/[1]两数之和/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """time O(n^2), space O(1), n 为 nums 长度"""
        for i in range(len(nums)):
            anotherValue = target - nums[i]
            # 列表 in 查询的时间复杂度为 0(n)
            if anotherValue in nums[i + 1 :]:
                return [i, nums.index(anotherValue, i + 1)]

        return []

File: algorithms/[1]两数之和/Solution.cs
// topics = ["数组", "哈希表"]

public class Solution
{
    /// <summary>
    /// Hash Table
    /// time O(n), space O(n), n 为 nums 长度
    /// </summary>
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> kvs = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int anotherValue = target - nums[i];
            if (kvs.ContainsKey(anotherValue))
            {
                return new int[] { kvs[anotherValue], i };
            }
            if (!kvs.ContainsKey(nums[i]))
            {
                kvs.Add(nums[i], i);
            }
        }

        return new int[] { -1, -1 };
    }
}
